### PR TITLE
Make TE tests transformerengine version agnostic

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Description
- grab transformerengine version from the container in order to find correct source code to test with

### Tests run
- successfully run test on [13cb111](https://github.com/aws/deep-learning-containers/pull/3354/commits/13cb111657048e6860eb381373a4f0589cbd9519)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/sagemaker.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
     # and then lets them handle their own workload. --dist=worksteal allows any processes that
     # finish their tests faster to steal tests from the queues for slower workers.
     remote_pytest_cmd = (
-        f"pytest -n=logical --dist worksteal -rA {integration_path} --region {region} "
+        f"pytest -n=logical --dist worksteal -rA {integration_path}/test_transformerengine.py --region {region} "
```
Comment (arjkesh) [inline test/test_utils/sagemaker.py:237]: temp change to only run TE test

File: test/dlc_tests/ec2/test_transformerengine.py [hunk 2]
```diff
@@ -40,9 +40,10 @@ def test_pytorch_transformerengine(
     pytorch_training, ec2_connection, region, ec2_instance_type, gpu_only, py3_only
 ):
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_training)
+    image_framework_version = Version(image_framework_version)
     image_cuda_version = get_cuda_version_from_tag(pytorch_training)
     if not (
-        Version(image_framework_version) in SpecifierSet(">=2.0") and image_cuda_version >= "cu121"
```
Comment (arjkesh) [inline test/dlc_tests/ec2/test_transformerengine.py:45]: change no longer necessary, will revert

File: pytorch/training/buildspec.yml [hunk 3]
```diff
@@ -50,7 +50,7 @@ images:
     tag_python_version: &TAG_PYTHON_VERSION py310
     cuda_version: &CUDA_VERSION cu121
     os_version: &OS_VERSION ubuntu20.04
-    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-ec2" ]
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-sagemaker" ]
```
Comment (arjkesh) [inline pytorch/training/buildspec.yml:53]: spoof SM image to run SM tests on TE

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_transformerengine.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2018-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_transformerengine.py:1]: update header

**Review by shantanutrip — APPROVED** [review]

**Pull request merged**